tic-tac-toe-app/App.jsx

```diff
@@ -1,5 +1,87 @@
+function Cells(props) {
+  return <button onClick={props.click} className='button'>{props.value}</button>
+}
+
+function Board(props) {
+  const [boardState, setBoardState] = React.useState(new Array(9).fill(null));
+
+  function declareWinner(board) {
+    const conditions = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [0, 4, 8],
+      [2, 4, 6],
+    ];
+
+    for (let i = 0; i < conditions.length; i++) {
+      const [first, second, third] = conditions[i]
+      if (board[first] && board[second] === board[first] && board[third] === board[first]) {
+        return board[first];
+      }
+    }
+    return null;
+  }
+
+  const winner = declareWinner(boardState)
+  let status;
+  if (winner) {
+    status = `The winner is ${winner}!`;
+  }
+
+  function clickHandler(num) {
+    if (!winner && !boardState[num]) {
+      boardState[num] = props.turn;
+      setBoardState(boardState)
+      props.setTurn((props.turn === 'X') ? 'O' : 'X');
```

Comment:
This is just a matter of preference: when using multiple props, the [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) can be used to help reduce code repetition.